geosnap/analyze/network.py

```diff
@@ -1,27 +1,50 @@
 import geopandas as gpd
-import numpy as np
+import pandana as pdna
 import pandas as pd
 from libpysal.cg import alpha_shape_auto
-from tqdm.auto import tqdm
+from shapely import concave_hull
+from shapely.geometry import MultiPoint
+
+
+def _geom_to_hull(geom, ratio, allow_holes):
+    return concave_hull(MultiPoint(geom.tolist()), ratio=ratio, allow_holes=allow_holes)
+
+
+def _geom_to_alpha(geom):
+    return alpha_shape_auto(geom.get_coordinates()[["x", "y"]].values)
+
+
+def _points_to_poly(df, column, algorithm="alpha", ratio=0.2, allow_holes=False):
+    if algorithm == "alpha":
+        output = df.groupby(column)["geometry"].apply(_geom_to_alpha)
+    elif algorithm == "hull":
+        output = df.groupby(column)["geometry"].apply(
+            lambda x: _geom_to_hull(x, ratio, allow_holes)
+        )
+    else:
+        raise ValueError(
+            f"`algorithm must be either 'alpha' or 'hull' but {algorithm} was passed"
+        )
+    return output
 
 
 def pdna_to_adj(origins, network, threshold, reindex=True, drop_nonorigins=True):
-    """Create an adjacency list of shortest network-based travel between origins
-       and destinations in a pandana.Network.
+    """Create an adjacency list of shortest network-based travel between
```

Comment:
Probably understandable here that "shortest network-based travel" is distance (as it is spelled out in the `threshold` description below), but also may not hurt to explicitly state "distance" (as opposed to travel time, etc.) here.

Reply:
impedance comes directly from the network, so we dont actually know what it is in the function. It's not necessarily distance

Reply:
Well then... never mind.